tests/plenary/async/async_spec.lua

```diff
@@ -0,0 +1,32 @@
+require('plenary.async').tests.add_to_env()
+
+describe('async', function()
+  a.it('void functions can call wrapped functions', function()
+    local stat = 0
+    local saved_arg
+
+    local wrapped = a.wrap(function(inc, callback)
+      stat = stat + inc
+      callback()
+    end, 2)
+
+    local voided = a.void(function(arg)
+      wrapped(1)
+      wrapped(2)
+      wrapped(3)
+      stat = stat + 1
+      saved_arg = arg
+    end)
+
+    voided('hello')
+
+    local function check()
+      return stat == 7 and saved_arg == 'hello'
+    end
+
+    if not vim.wait(2000, check, 20, false) then
```

Comment:
why don't you just do a direct assert

Reply:
I naturally treat void functions as non-blocking, but yeah, I guess `voided` here is essentially blocking.